pkg/telemetry/statsworker.go

```diff
@@ -15,158 +15,178 @@ type StatsWorker struct {
 	roomName      livekit.RoomName
 	participantID livekit.ParticipantID
 
-	drainStats       map[livekit.TrackID]bool
-	outgoingPerTrack map[livekit.TrackID]*Stats
-	incomingPerTrack map[livekit.TrackID]*Stats
+	outgoingPerTrack map[livekit.TrackID][]*livekit.AnalyticsStat
+	incomingPerTrack map[livekit.TrackID][]*livekit.AnalyticsStat
 }
 
-type Stats struct {
```

Comment:
Removing diffing stats in this file. This was also (I think) was written to have current and next (or current and previous) which assumed that this runs at less than or equal to two times the interval of connection stats. In this change, the stats from connection stats is just maintained in a list and the list for each track is emptied when the stats worker runs. The list of stats for a track is coalesced in the function `coalesce()`.

Also, there is no need to maintain drain tracks as on every run of stats worker will coalesce the queued up stats, post and clean up.

Reply:
Okay, looks like some diffing is needed as the tests are failing. I have not looked at the details of the failures, but can see it is possibly due to not diffing.

@shishirng Which of these stats should be posted as delta values to the telemetry service?